src/network/ethereum.rs

```diff
@@ -1,87 +1,143 @@
 //! Ethereum
-use std::collections::HashMap;
+use std::{collections::HashMap, convert::Infallible};
 
+use alloy_chains::NamedChain;
 use alloy_consensus::TxType;
 use alloy_primitives::U256;
 use alloy_rpc_types::{Header, Transaction};
-use revm::primitives::{BlockEnv, SpecId};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
 
 use crate::{
     mv_memory::{LazyAddresses, MvMemory},
-    primitives::TransactionParsingError,
     BuildIdentityHasher, MemoryLocation, TxIdx,
 };
 
-/// Build MvMemory
-pub(crate) fn build_mv_memory(
-    hasher: &ahash::RandomState,
-    block_env: &BlockEnv,
-    block_size: usize,
-) -> MvMemory {
-    let beneficiary_location_hash = hasher.hash_one(MemoryLocation::Basic(block_env.coinbase));
-
-    // TODO: Estimate more locations based on sender, to, etc.
-    let mut estimated_locations = HashMap::with_hasher(BuildIdentityHasher::default());
-    estimated_locations.insert(
-        beneficiary_location_hash,
-        (0..block_size).collect::<Vec<TxIdx>>(),
-    );
-
-    let mut lazy_addresses = LazyAddresses::default();
-    lazy_addresses.0.insert(block_env.coinbase);
-
-    MvMemory::new(block_size, estimated_locations, lazy_addresses)
+use super::PevmChain;
+
+/// Implementation of [PevmChain] for Ethereum
+#[derive(Debug, Clone)]
+pub struct PevmChainEthereum {
+    id: u64,
```

Comment:
Not yet important but this not being `pub` without a `new` function would make it impossible to custom.

Reply:
Adding `PevmChainEthereum::new()` will require to change `get_block_spec`.

I will add a comment to state that currently, only Ethereum Mainnet is supported.